pulp_service/pulp_service/app/middleware.py

```diff
@@ -0,0 +1,57 @@
+from contextlib import suppress
+import cProfile
+import logging
+import marshal
+import tempfile
+
+from django.db import IntegrityError
+from django.utils.deprecation import MiddlewareMixin
+
+from pulpcore.app.models import Artifact
+from pulpcore.app.util import get_artifact_url
+
+_logger = logging.getLogger(__name__)
+
+class ProfilerMiddleware(MiddlewareMixin):
+    """
+    Simple profile middleware to profile django views. To run it, add
+    x-profile-request header to the request.
+
+    This is adapted from an example found here:
+    https://github.com/omarish/django-cprofile-middleware/blob/master/django_cprofile_middleware/middleware.py
+    """
+    PROFILER_REQUEST_ATTR_NAME = '_django_cprofile_middleware_profiler'
+
+    def can(self, request):
+        if 'HTTP_X_PROFILE_REQUEST' in request.META:
```

Comment:
So the first draft is entirely yes/no based on the presence of the header? I'm ok with that if so.

Reply:
that is correct